### PR TITLE
Readded TileLoader.NearbyEffects

### PR DESCRIPTION
**Pull request by Agrair**
Too lazy to add anything proper. I didn't actually readd the `closer` bool because vanilla properly split it up now.

### CODE REVIEW

**Review by Chicken-Bones — APPROVED** [review]
Everything checks out from a code perspective. If any other dev (@Mirsario @JavidPack) can verify the locations of the hook calls, just click merge.

File: patching_todo.txt [hunk 1]
```diff
@@ -112,16 +112,9 @@
...
-		//Re-add TileLoader.NearbyEffects calls (perhaps multiple) to somewhere in SceneMetrics
-		
 		//Re-add this MusicBox logic, to somewhere in SceneMetrics:
 			if (TileLoader.IsModMusicBox(tile) && tile.frameX >= 36)
 				Main.musicBox = SoundLoader.tileToMusic[tile.type][tile.frameY / 36 * 36];
-		
-		//Call this at the end of SceneMetrics.ScanAndExportToMain:
-			WorldHooks.TileCountsAvailable(_tileCounts);
-		
-		//Resize the arrays that ScreenMetrics.UpdateOreFinderData uses.
```
Comment (Agrair) [inline patching_todo.txt:124]: If anyone is curious why I deleted this line even though I didn't patch it, it was already done way back on 1.3, `UpdateOreFinderData` just uses the same arrays as before.

**Comment by Agrair** [comment]
Also, I noticed that `SceneMetrics._tileCount` is resized by `TileLoader` with the `Resize(arr, nextTile)` pattern, but `SceneMetrics.NPCBannerBuff` is resized by `SceneMetrics` itself after all mods have been loaded, and only if npcs were added, making it probably far more performant. What's up with that?

**Review by Mirsario — APPROVED** [review]

**Comment by Mirsario** [comment]
> 
> 
> Also, I noticed that `SceneMetrics._tileCount` is resized by `TileLoader` with the `Resize(arr, nextTile)` pattern, but `SceneMetrics.NPCBannerBuff` is resized by `SceneMetrics` itself after all mods have been loaded, and only if npcs were added, making it probably far more performant. What's up with that?

Not sure. I don't remember if I'm the one who worked on that.

**Pull request merged**